File: Submissions.CSharp/Challenges/FizzBuzz.cs
namespace LeetCode;

using FluentAssertions;
using NUnit.Framework;

public sealed partial class Submission
{
    [LeetCode("Fizz Buzz", Difficulty.Easy, Category.NotInNeetCode)]
    public static IList<string> FizzBuzz(int n)
    {
        var result = new List<string>();

        for (int i = 1; i <= n; i++)
        {
            var value = string.Empty;

            if (i % 3 == 0)
            {
                value = "Fizz";
            }

            if (i % 5 == 0)
            {
                value += "Buzz";
            }

            if (string.IsNullOrEmpty(value))
            {
                value = i.ToString();
            }

            result.Add(value);
        }

        return result;
    }

    [Test]
    public void FizzBuzzTest()
    {
        var input1 = 3;
        var output1 = new List<string> { "1", "2", "Fizz" };
        var input2 = 5;
        var output2 = new List<string> { "1", "2", "Fizz", "4", "Buzz" };
        var input3 = 15;
        var output3 = new List<string> { "1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8", "Fizz", "Buzz", "11", "Fizz", "13", "14", "FizzBuzz" };

        FizzBuzz(input1).Should().Equal(output1);
        FizzBuzz(input2).Should().Equal(output2);
        FizzBuzz(input3).Should().Equal(output3);
    }
}

File: Submissions.FSharp/Challenges/FizzBuzz.fs
module LeetCode.FizzBuzz

open FsUnit
open NUnit.Framework

[<LeetCode("Fizz Buzz", Difficulty.Easy, Category.NotInNeetCode)>]
let FizzBuzz n =
    [1 .. n] |>
    // TODO: Append Buzz to Fizz instead multiple operations
    Seq.map (function    
        | x when x % 5 = 0 && x % 3 = 0 -> "FizzBuzz"
        | x when x % 3 = 0 -> "Fizz"
        | x when x % 5 = 0 -> "Buzz"
        | x -> string x)

[<Test>]
let ``When passing 3 FizzBuzz should return expected result`` () =
    FizzBuzz(3) |> should equal ["1"; "2"; "Fizz"]

[<Test>]
let ``When passing 5 FizzBuzz should return expected result`` () =
    FizzBuzz(5) |> should equal ["1"; "2"; "Fizz"; "4"; "Buzz"]

[<Test>]
let ``When passing 15 FizzBuzz should return expected result`` () =
    FizzBuzz(15) |> should equal ["1"; "2"; "Fizz"; "4"; "Buzz"; "Fizz"; "7"; "8"; "Fizz"; "Buzz"; "11"; "Fizz"; "13"; "14"; "FizzBuzz"]